### PR TITLE
Accessible Names Practice: Recommend association via 'for' attribute when using 'label' element

### PR DESCRIPTION
**Pull request by mcking65**
Inspired by #2871, this is an alternative proposal for resolving #2870.

@patrickhlauke, I originally was looking at writing this as a suggestion in #2871, but it became too complicated. I appreciate the time you took to make a PR; I do not intend to be dismissive of your work on the issue.

In addition to reducing word count, here are the notable editorial differences and my reasons for proposing them. Feedback is welcome on all of them.

1. It describes the preferred syntax first. It seems beneficial to present the most valuable and important information first and enable people to skip over information they may not use.
2. It uses more concrete language for explaining the reason for the preference. In particular, help people understand that it is due to specific AT/Browser combinations not calculating the correct name.
3. It emphasizes the preference by mentioning it two times.
#### Preview

[Preview of the proposed changes to the names and description practice](https://deploy-preview-283--aria-practices.netlify.app/aria/apg/practices/names-and-descriptions/#namingtechniques)

___
[WAI Preview Link](https://deploy-preview-283--aria-practices.netlify.app/ARIA/apg) _(Last built on Sun, 17 Dec 2023 08:02:10 GMT)._

### CODE REVIEW

File: content/practices/names-and-descriptions/names-and-descriptions-practice.html
Comment (patrickhlauke) [inline content/practices/names-and-descriptions/names-and-descriptions-practice.html:311]:
```suggestion
            The syntax with the broadest browser and assistive technology support is to set the <code>for</code> attribute on the <code>label</code> element to the <code>id</code> of the control.
```

File: content/practices/names-and-descriptions/names-and-descriptions-practice.html
Comment (patrickhlauke) [inline content/practices/names-and-descriptions/names-and-descriptions-practice.html:318]: the "in as many" sounds like we actually went around counting the totals. i'd generalise it more into something more like... "The other syntax is to wrap […]. However, this syntax is not supported by all combinations of assistive technologies and browsers." (if possible, i'd even be more specific and say which ones, at time of writing, have problems)

**Comment by patrickhlauke** [comment]
while the implicit/explicit naming isn't in the HTML spec per se, it's a fairly common way of referring to these two syntaxes https://www.google.com/search?q=implicit+explicit+html+label ... including in actual WAI material https://www.w3.org/WAI/tutorials/forms/labels/#:~:text=Associating%20labels%20implicitly,-In%20some%20situations&text=In%20this%20case%2C%20the%20label,better%20supported%20by%20assistive%20technology.

**Comment by mcking65** [comment]
@howard-e @ccanash 

The preview link is still missing on this PR even though none of the checks are failing. Is the issue with the build workflow resolved? Is it possible to provide a preview for this PR?

File: content/practices/names-and-descriptions/names-and-descriptions-practice.html
Comment (mcking65) [reply content/practices/names-and-descriptions/names-and-descriptions-practice.html]: I made some revisions to address this. Rather than just saying it is not supported, I tried to use words that help the reader understand what is not happening. In general, we avoid naming AT in the APG. We are plannig to provide support data that can show AT specifics via the aria-at data using the AT support tables on the example pages. Over time, that data will be able to surface in more places.

**Comment by mcking65** [comment]
@patrickhlauke, @a11ydoer, There is a possibility we will do a publication this week (Dec 12). If so, I'll have to merge this tomorrow (Dec 11) morning if hthis were to be included. Otherwise, it can be published on Dec 19.

I'll wait for approving reviews from both of you before merging.

**Comment by a11ydoer** [comment]
@mcking65 I reviewed the changes, and it looks good. I fully understand @patrickhlauke's kind intention to help readers with non-supported browsers and AT combinations, but it would be hard to update all the details as technology advances. Furthermore, the AT project will fulfill that need.

File: content/practices/names-and-descriptions/names-and-descriptions-practice.html
Comment (a11ydoer) [inline content/practices/names-and-descriptions/names-and-descriptions-practice.html:1]: @mcking65 this is easy to read and understand. Thanks for the update.

**Comment by patrickhlauke** [comment]
i would still like to see the "implicit vs explicit" mention though, based on https://github.com/w3c/aria-practices/pull/2882#issuecomment-1843557219 ... developers do use that terminology, so why try to avoid it for some "the spec itself doesn't call it that"?

**Comment by a11ydoer** [comment]
> i would still like to see the "implicit vs explicit" mention though, based on [#2882 (comment)](https://github.com/w3c/aria-practices/pull/2882#issuecomment-1843557219) ... developers do use that terminology, so why try to avoid it for some "the spec itself doesn't call it that"?

Hi @patrickhlauke, 

Actually, I suggested a similar thing at the APG meeting last week. Use "implicit vs. explicit" concepts and link those definitions to somewhere, even in the ARIA spec. I see that WAI material used implicit and explicit concepts from your WAI reference. Can you help us to find the definitions of "explicit" and "implicit" so that we can use the words?

All the reasoning behind this is that APG is aiming for easy reading, targeting a 6th-grade reading level, aligning with WCAG 3 efforts. This approach is intended for **both developers and beginners in ARIA**. I've checked, and it seems that "implicit" and "explicit" may not be at a 6th-grade vocabulary level.

**Comment by mcking65** [comment]
@a11ydoer wrote:

> All the reasoning behind this is that APG is aiming for easy reading, targeting a 6th-grade reading level, aligning with WCAG 3 efforts. This approach is intended for **both developers and beginners in ARIA**.

I didn't realize that we had 6th-grade level is a target. I thought that there was so much dispute over the meaning of such measurement systems and the bias built into them that we were explicitly not adopting them.

I have simply been striving for clear language that is not more complex than necessary.

>I've checked, and it seems that "implicit" and "explicit" may not be at a 6th-grade vocabulary level.

I'm a little confused. Is this reason not to use the words "implicit" and "explicit"? If we were to add links to their definitions, it wouldn't change the grade-level of the prose.

The APG text itself provides the special contextual meaning of "implicit labeling" and "explicit labeling". So, I suggest that if we add these terms to the text that we would not add definition links. The links would be more distracting than helpful because they would point to another place that provides essentially the same information that is in the APG prose.

I'm not sure I understand the value of including the implicit/explicit terminology, but I will add it and people can decide if the content is or is not better with that terminology included.

**Comment by patrickhlauke** [comment]
i'll admit that i seriously doubt some of the more inherently complex and technical concepts of ARIA will be (easily or at all) conveyable in simple language.

in terms of adding a definition, I was thinking along the lines of my original PR - where the concept itself is explained, and then at the end adding `This is often called <em>implicit</em> labeling`. This, I'd argue, then doesn't need it own dictionary definition of what "implict" means, because if a reader doesn't understand/know that term, they still understand the concept because that explanation was not reliant on it.

**Comment by shawna-slh** [comment]
+1 for keeping the language as simple as feasible
+1 for not getting too "hung up" on calculated reading levels
+1 to where the concept itself is explained, and then at the end adding This is often called <em>implicit</em> labeling. -- for us old birds for whom implicit and explicit labels makes sense :-)

thanks for tagging me to weigh in

**Comment by mcking65** [comment]
I updated the text to include the terms explicit association and implicit association.

**Review by daniel-montalvo — APPROVED** [review]

File: content/practices/names-and-descriptions/names-and-descriptions-practice.html
Comment (patrickhlauke) [inline content/practices/names-and-descriptions/names-and-descriptions-practice.html:310]: If we're aiming for simplicity/understandability (regardless of reading level etc), maybe instead of referring to "syntaxes" we could just talk about "ways" here and subsequently

File: content/practices/names-and-descriptions/names-and-descriptions-practice.html
Comment (patrickhlauke) [inline content/practices/names-and-descriptions/names-and-descriptions-practice.html:320]: might be prudent to qualify this statement with "Currently, some combinations..." or similar, as the situation may well change

File: content/practices/names-and-descriptions/names-and-descriptions-practice.html
Comment (a11ydoer) [reply content/practices/names-and-descriptions/names-and-descriptions-practice.html:310]: Thanks for considering understandability and simplicity, @patrickhlauke !

Comment (mcking65) [reply content/practices/names-and-descriptions/names-and-descriptions-practice.html]: Revised acordingly.

File: content/practices/names-and-descriptions/names-and-descriptions-practice.html
Comment (mcking65) [reply content/practices/names-and-descriptions/names-and-descriptions-practice.html:320]: If the bugs are fixed and this text is still present, the text will be equally inaccurate whether or not it includes the word "currently". The only way to avoid that problem would be to include a phrase like "as of December 2023", but I'm reluctant to start including such temporal prose in the APG.

So, it is somewhat risky to include any information of this nature. We have tried to avoid doing so. The long term strategy for addressing AT support is interop data from [aria--at](https://aria-at.w3.org/)., which  will be updated with each new release of a covered AT

**Review by patrickhlauke — APPROVED** [review]

**Pull request merged**
